lib/topological_inventory/providers/common/save_inventory/saver.rb

```diff
@@ -11,10 +11,10 @@ class Saver
           # it 200 bytes, just for sure.
           KAFKA_RESERVED_HEADER_SIZE = 200
 
-          def initialize(client:, logger:, max_bytes: 1_000_000)
+          def initialize(client:, logger:, max_bytes:)
```

Comment:
Removing the default for `max_bytes` forces the caller to always pass a value which breaks the current caller.  I think you want to leave this value as-is and limit it with the `clamp` call below.

Reply:
I would just decrease this default value and keep other unchanged. 

Anyway the Kafka size is 1MB it means 1 048 576B (regarding to @bzwei) so there should be enough space for all the additional headers. So I guess our core of problem is somewhere else